### PR TITLE
fix: Merge user preferences with default preferences

### PR DESCRIPTION
**Pull request by JeangelLF**
## Ticket

https://github.com/corona-school/project-user/issues/1236

## What was done?

- Merge stored with default preferences on field resolver, to display the actual preferences state in the UI correctly

### CODE REVIEW

File: graphql/user/fields.ts [hunk 1]
```diff
@@ -155,7 +155,8 @@ export class UserFieldsResolver {
     @FieldResolver((returns) => JSONResolver, { nullable: true })
     @Authorized(Role.OWNER, Role.ADMIN)
     async notificationPreferences(@Root() user: User) {
-        return (await queryUser(user, { notificationPreferences: true })).notificationPreferences ?? DEFAULT_PREFERENCES;
+        const storedPreferences = (await queryUser(user, { notificationPreferences: true })).notificationPreferences;
+        return storedPreferences ? Object.assign(storedPreferences, DEFAULT_PREFERENCES) : DEFAULT_PREFERENCES;
```
[the remark this answers is not in the document]
Comment (JeangelLF) [reply graphql/user/fields.ts:159]: > This mutates storedPrefereces? Would prefer Object.assign({}, ...) or { a..., b... }

You're right. I also remembered we have a nested object and some keys may or may not exist. So I think `_.merge` should be fine in this case

Bildschirmfoto 2024-06-10 um 18 25 34

> Also dont we have a function for this in common/notification?

I just checked and I didn't find it

[the remark this answers is not in the document]
Comment (Jonasdoubleyou) [reply graphql/user/fields.ts]: Well then we should create one. If notification system and GraphQL did diverge, adding two different implementations to derive them does not solve the problem?

[the remark this answers is not in the document]
Comment (JeangelLF) [reply graphql/user/fields.ts]: Alright, I'll add a function for it. 
The only difference is that the notification system needs the `ENABLED_NOTIFICATIONS` object _(via `ALL_PREFERENCES`)_ and the GraphQL part just needs the `DEFAULT_PREFERENCES` object.

Could we agree on just using `ALL_PREFERENCES`?

[the remark this answers is not in the document]
Comment (JeangelLF) [reply graphql/user/fields.ts]: Alright, I've added it

**Review by Jonasdoubleyou — APPROVED** [review]

**Pull request merged**